bindgen/ir/TypeDef.cpp

```diff
@@ -46,4 +46,27 @@ bool TypeDef::operator==(const Type &other) const {
     return false;
 }
 
-std::shared_ptr<Location> TypeDef::getLocation() const { return location; }
+std::shared_ptr<Location> TypeDef::getLocation() const {
+    if (!type) {
+        assert(false && "This method should not be called on typedef that "
+                        "references an opaque type because location cannot be "
+                        "defined.");
+    }
+    if (location) {
+        /* if typedef is not generated */
+        return location;
+    }
+    auto structOrUnionPointer =
+        std::dynamic_pointer_cast<const StructOrUnion>(type);
+    if (structOrUnionPointer) {
+        return structOrUnionPointer->getLocation();
+    }
+    auto enumPointer = std::dynamic_pointer_cast<const Enum>(type);
+    if (enumPointer) {
+        return enumPointer->getLocation();
+    }
+    assert(false &&
+           "Generated typedef may reference only struct, union or enum");
```

Comment:
Since assertions can be deactivated at compile time (at least in C) I am not sure they are a good means of error reporting.

Reply:
Okay, it's better to throw an exception